src/consumer.coffee

```diff
@@ -1,7 +1,52 @@
+crypto            = require 'crypto'
+URL         = require 'url-parse'
```

Comment:
Reason for using _url-parse_ over the built-in [url](https://nodejs.org/api/url.html)?